frontend/lib/src/features/sign_up/sign_up_step_two.dart

```diff
@@ -1,134 +1,203 @@
+import 'dart:convert';
+import 'dart:io';
+
 import 'package:flutter/material.dart';
 import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
 import 'package:form_builder_validators/form_builder_validators.dart';
+import 'package:flow_builder/flow_builder.dart';
 import 'package:get/get.dart';
 
-import 'package:frontend/src/features/sign_up/components/breadcrumb.dart';
+import 'package:frontend/src/helper/dio.dart';
+import 'package:frontend/src/helper/storage.dart';
+import 'package:frontend/src/features/home_screen.dart';
 import 'package:frontend/src/components/input/input_field.dart';
 import 'package:frontend/src/components/button.dart';
-import 'package:frontend/src/features/login/login_page.dart';
-import 'package:frontend/src/components/auth/auth_header.dart';
+import 'package:frontend/src/features/sign_up/sign_up_flow.dart';
+import 'package:frontend/src/controllers/sign_up_controller.dart';
+import 'package:frontend/src/components/breadcrumb.dart';
 
 /*
   The page where user can fill the second page of sign up.
+
+  @param, userFormData, the current sign up flow data.
 */
 class SignUpStepTwo extends StatelessWidget {
-  const SignUpStepTwo({super.key});
+  const SignUpStepTwo({super.key, required this.userFormData});
+
+  final UserFormData userFormData;
 
   @override
   Widget build(BuildContext context) {
+    final SignUpController signUpController = Get.put(SignUpController());
     final formKey = GlobalKey<FormBuilderState>();
+    const storage = FlutterSecureStorage();
+
+    void onSubmit() async {
+      signUpController.setSignUpSecondButtonEnabled = false;
+
+      await NetworkConfig().getCsrftoken();
+      final client = NetworkConfig().client;
+
+      await storage.delete(key: StorageKeys.loginToken.name);
+      await storage.delete(key: StorageKeys.userId.name);
+
+      try {
+        final signUpResponse = await client.post(
+          authUrl,
+          data: {
+            'first_name': userFormData.firstName,
+            'middle_name': userFormData.middleName,
+            'last_name': userFormData.lastName,
+            'address': userFormData.address,
+            'birthday': userFormData.birthday.toString(),
+            'user_name': formKey.currentState!.fields['username']!.value,
+            'email': formKey.currentState!.fields['email']!.value,
+            'password': formKey.currentState!.fields['password']!.value,
+            'password_confirmation':
+                formKey.currentState!.fields['password_confirmation']!.value,
+          },
+        );
+
+        if (signUpResponse.statusCode == HttpStatus.ok) {
+          final loginToken = signUpResponse.data['data']['token'];
+          final userId = signUpResponse.data['data']['id'];
+
+          storage.write(key: StorageKeys.loginToken.name, value: loginToken);
```

Comment:
I removed the `storage.write()`, since after creating the account successfully is will navigate to login page and the login page will handle the `storage.write()` to store the loginToken and userId.